@shared/api/external.ts

```diff
@@ -132,3 +133,59 @@ export const requestConnectionStatus = async (): Promise<boolean> => {
 
   return response.isConnected;
 };
+
+export const requestAllowedStatus = async (): Promise<boolean> => {
+  let response = {
+    isAllowed: false,
+  };
+
+  try {
+    response = await sendMessageToContentScript({
+      type: EXTERNAL_SERVICE_TYPES.REQUEST_ALLOWED_STATUS,
+    });
+  } catch (e) {
+    console.error(e);
+  }
+
+  return response.isAllowed;
+};
+
+export const setAllowedStatus = async (): Promise<boolean> => {
+  let response = {
+    isAllowed: false,
+    error: "",
+  };
+
+  try {
+    response = await sendMessageToContentScript({
+      type: EXTERNAL_SERVICE_TYPES.SET_ALLOWED_STATUS,
+    });
+  } catch (e) {
+    console.error(e);
+  }
+
+  const { isAllowed, error } = response;
+
+  if (error) {
+    throw error;
+  }
+  return isAllowed;
+};
+
+export const requestUserInfo = async (): Promise<UserInfo> => {
```

Comment:
get userinfo (includes public key) in a way that doesn't prompt the user. It just returns an empty string if the dapp isn't allowed or the user needs to re-auth